lib/pagseguro/collection_base.rb

```diff
@@ -0,0 +1,41 @@
+module PagSeguro
+  class CollectionBase
+    extend CollectionManagers::MacroMethods
+    extend Forwardable
+    include Enumerable
+
+    def_delegators :@store, :size, :clear, :empty?, :any?, :each
+
+    def initialize
+      @store = []
+    end
+
+    def collection_manager
+      CollectionManagers::GenericManager
+    end
+
+    def collection_type
+      eval singularize(self.class.to_s)
```

Comment:
É por conta que no ruby 1.9.3 se tu tem esse esquema de classe:

```class A
  class B
  end
end```

Então você deve usar `A.const_get('B')`

Porém se você utiliza o self.class.to_s retornará `A::B`, então quando for utilizar `A.const_get('A::B')` apresentará um erro.

A solução foi da um split pegando o último elemento que no caso iria ser o `B`